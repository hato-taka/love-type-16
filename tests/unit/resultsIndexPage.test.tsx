import React from 'react';
import { describe, expect, it, vi, beforeEach } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import type { PersonalityResult } from '@/types/quiz';

const mockGetAllResults = vi.fn((): PersonalityResult[] => []);

vi.mock('next/image', () => ({
  __esModule: true,
  default: (props: { src: string; alt: string }) => {
    const { src, alt, ...rest } = props as Record<string, unknown>;
    const { fill: _fill, sizes: _sizes, unoptimized: _unoptimized, ...imgProps } = rest;

    return (
      // eslint-disable-next-line @next/next/no-img-element
      <img src={src as string} alt={alt as string} data-testid="mock-next-image" {...imgProps} />
    );
  }
}));

vi.mock('next/link', () => ({
  __esModule: true,
  default: ({
    href,
    children,
    prefetch: _prefetch,
    ...rest
  }: {
    href: string;
    children: React.ReactNode;
    prefetch?: boolean;
  }) => (
    <a href={href} {...rest}>
      {children}
    </a>
  )
}));

vi.mock('@/lib/quiz', () => ({
  getAllResults: () => mockGetAllResults()
}));

// モックデータ生成ヘルパー
function createResult(overrides: Partial<PersonalityResult> = {}): PersonalityResult {
  return {
    type: 'SCA-T',
    character: 'テストキャラ',
    title: 'テストタイトル',
    summary: 'キャッチーな要約テキスト',
    description: {
      workStyle: '仕事観',
      loveView: '恋愛観',
      marriageView: '結婚観',
      lifeView: '人生観',
      oshiStyle: '推し活観'
    },
    image: 'results/test.png',
    shareImageUrl: 'https://cdn.example.com/test.png',
    ...overrides
  };
}

beforeEach(() => {
  mockGetAllResults.mockReset();
});

describe('ResultsIndexPage', () => {
  it('renders stable markup for the results collection', async () => {
    mockGetAllResults.mockReturnValue([
      createResult({ type: 'SAA-T', character: 'キキ', summary: '冒険心旺盛なクリエイター' }),
      createResult({
        type: 'IAF-Q',
        character: 'あひるのペックル',
        shareImageUrl: undefined,
        image: 'results/pekkle.png'
      })
    ]);

    const { default: ResultsIndexPage } = await import('@/app/results/page');
    const html = renderToStaticMarkup(<ResultsIndexPage />);

    expect(html).toMatchSnapshot();
  });
});
