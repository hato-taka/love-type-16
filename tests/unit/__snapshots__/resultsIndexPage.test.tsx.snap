// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResultsIndexPage > renders stable markup for the results collection 1`] = `"<div class="relative mx-auto my-6 w-full max-w-3xl overflow-hidden rounded-[26px] bg-white/90 px-5 py-8 shadow-pop sm:my-12 sm:px-8 sm:py-10 lg:max-w-4xl max-w-5xl px-5 py-8 sm:px-10 sm:py-12"><div class="pointer-events-none absolute -left-28 -top-32 h-64 w-64 rounded-full bg-candy-base/20 blur-3xl" aria-hidden="true"></div><div class="pointer-events-none absolute -bottom-36 -right-28 h-72 w-72 rounded-full bg-candy-accent/20 blur-3xl" aria-hidden="true"></div><div class="relative z-10 space-y-10"><header class="space-y-3 text-center"><p class="text-xs font-semibold uppercase tracking-[0.3em] text-candy-lavender">Personality Library</p><h1 class="text-[clamp(2rem,7vw,2.8rem)] font-semibold text-candy-plum">全16タイプのサンリオキャラ診断一覧</h1><p class="mx-auto max-w-3xl text-sm leading-relaxed text-candy-lavender">診断タイプごとのキャラクター、肩書き、サマリーをまとめています。興味のあるタイプを選んで詳細ページで性格傾向を深掘りしましょう。</p></header><div class="grid gap-6 sm:grid-cols-2 lg:grid-cols-3"><a href="/results/iaf-q" class="group flex h-full flex-col gap-4 rounded-3xl bg-white/90 p-5 text-left shadow-pop transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-candy-base/60 hover:-translate-y-0.5 hover:shadow-popHover"><div class="relative aspect-square w-full overflow-hidden rounded-2xl bg-gradient-to-br from-[#fff7fb] via-[#ffe3f1] to-[#d7f2ff]"><img src="/results/pekkle.png" alt="あひるのペックル のイラスト" data-testid="mock-next-image" class="object-contain transition-transform duration-200 group-hover:scale-[1.04]"/></div><div class="space-y-2"><span class="inline-flex items-center rounded-full bg-candy-base/10 px-4 py-1 text-xs font-semibold uppercase tracking-[0.3em] text-candy-lavender">IAF-Q</span><h2 class="text-xl font-semibold leading-tight text-candy-plum">あひるのペックル</h2><p class="text-sm leading-relaxed text-candy-lavender/90">キャッチーな要約テキスト</p></div></a><a href="/results/saa-t" class="group flex h-full flex-col gap-4 rounded-3xl bg-white/90 p-5 text-left shadow-pop transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-candy-base/60 hover:-translate-y-0.5 hover:shadow-popHover"><div class="relative aspect-square w-full overflow-hidden rounded-2xl bg-gradient-to-br from-[#fff7fb] via-[#ffe3f1] to-[#d7f2ff]"><img src="https://cdn.example.com/test.png" alt="キキ のイラスト" data-testid="mock-next-image" class="object-contain transition-transform duration-200 group-hover:scale-[1.04]"/></div><div class="space-y-2"><span class="inline-flex items-center rounded-full bg-candy-base/10 px-4 py-1 text-xs font-semibold uppercase tracking-[0.3em] text-candy-lavender">SAA-T</span><h2 class="text-xl font-semibold leading-tight text-candy-plum">キキ</h2><p class="text-sm leading-relaxed text-candy-lavender/90">冒険心旺盛なクリエイター</p></div></a></div></div></div>"`;
